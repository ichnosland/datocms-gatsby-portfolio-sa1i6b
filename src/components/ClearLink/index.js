/**
*
* ClearLink
*
*/

import styled, { css } from 'styled-components';
import { Link } from 'react-router-dom';

export const clearLinkStyle = css`
  color: inherit;
  display: inherit;
  text-decoration: inherit;
`;

export const ClearLink = styled(Link)`
  ${clearLinkStyle}
`;

export default ClearLink;
