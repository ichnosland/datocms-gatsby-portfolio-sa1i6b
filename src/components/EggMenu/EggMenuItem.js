/**
*
* EggMenuItem
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Div from 'components/Div';
import ClearLink from 'components/ClearLink';
import Svg from 'components/Svg';
import { colore } from 'style/color';

const labelsPaddingVertical = '4px';
const labelsPaddingHorizontal = '10px';
const labelsFontSize = '16px';
const labelHoverOff = '0.5s';

export const EggItem = styled(Div)`
  position: relative;
  height: 30px;
  width: 30px;
  border-radius: 100%;
  background-color: ${colore.ui.contrast};
  overflow: visible;
`;

export const EggLink = styled(ClearLink)`
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: -1px;
  left: -2px;
  width: ${(props) => props.theme.menu.itemWidth};
  height: ${(props) => props.theme.menu.itemHeight};
  border: ${(props) => props.theme.menu.itemBorder ? props.theme.menu.itemBorder : 'none'};
  background-color: ${(props) => props.iconbg ? props.iconbg : colore.ui.contrast};
  outline: none;
  padding: 0;
  border-radius: ${(props) => props.theme.menu.linkRadius};
  box-shadow: 0 1px 2px rgba(0,0,0,.3);
  -webkit-user-drag: none;
  svg {
    fill: ${(props) => props.fill ? props.theme.brand : props.fill};
    max-width: 32px;
    max-height: 32px;
  }
`;

EggLink.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    menu: {
      itemBorder: 'none',
      itemWidth: '32px',
      itemHeight: '32px',
    },
  },
};

export const EggLinkLabel = styled.span`
  display: block;
  position: absolute;
  top: 3%;
  right: 45px;
  box-shadow: 0 1px 2px rgba(0,0,0,.3);
  background-color: ${colore.ui.contrast};
  padding-top: ${labelsPaddingVertical};
  padding-bottom: ${labelsPaddingVertical};
  padding-left: ${labelsPaddingHorizontal};
  padding-right: ${labelsPaddingHorizontal};
  border-radius: 6px;
  pointer-events: none;
  margin-top: - (${labelsPaddingVertical} + ${labelsFontSize} / 2);
  transition: all ${labelHoverOff};
`;

export default class EggMenuItem extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { url, titolo, icona, iconbg, fill, onOpen } = this.props;
    return (
      <EggItem>
        {(url || onOpen) &&
          <EggLink
            to={url || '#'}
            onClick={onOpen}
            iconbg={iconbg}
          >
            <Svg {...icona} fill={fill} />
            <EggLinkLabel>{titolo}</EggLinkLabel>
          </EggLink>
        }
      </EggItem>
    );
  }
}

EggMenuItem.propTypes = {
  url: PropTypes.string,
  titolo: PropTypes.string.isRequired,
  icona: PropTypes.object.isRequired,
  iconbg: PropTypes.string,
  fill: PropTypes.string,
  onOpen: PropTypes.func,
};
