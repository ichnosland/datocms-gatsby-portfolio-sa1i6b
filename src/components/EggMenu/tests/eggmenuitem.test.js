import React from 'react';
import { shallow } from 'enzyme';

import ClearLink from 'components/ClearLink';
import icon from 'icons/globals';
import EggMenuItem, { EggItem, EggLink, EggLinkLabel } from '../EggMenuItem';

const icone = 'feedback';
const theme = {
  radius: {
    buttons: '50px',
  },
  menu: {
    itemBorder: '2px solid green',
    itemWidth: '32px',
    itemHeight: '32px',
  },
};

describe('<EggMenuItem />', () => {
  it('counts EggMenuItem with one child and no url attribute', () => {
    const renderedComponent = shallow(
      <EggMenuItem
        titolo="Questo è un titolo"
        icona={icon[icone]}
        fill="#000"
        theme={theme}
      />
    );
    expect(renderedComponent.find(EggLink).length).toBe(0);
    expect(renderedComponent.find(EggItem).length).toBe(1);
    expect(renderedComponent.find(EggLinkLabel).length).toBe(0);
    expect(renderedComponent).toMatchSnapshot();
  });
  it('counts EggMenuItem children with url attribute', () => {
    const renderedComponent = shallow(
      <EggMenuItem
        titolo="Questo è un titolo"
        icona={icon[icone]}
        fill="#000"
        url="/path"
        onOpen={() => { }}
        theme={theme}
      />
    );
    expect(renderedComponent.find(EggItem).length).toBe(1);
    expect(renderedComponent.find(EggLink).length).toBe(1);
    expect(renderedComponent.find(EggLinkLabel).length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });
  it('counts EggMenuItem children with empty url attribute', () => {
    const renderedComponent = shallow(
      <EggMenuItem
        titolo="Questo è un titolo"
        icona={icon[icone]}
        fill="#000"
        onOpen={() => { }}
        theme={theme}
      />
    );
    expect(renderedComponent.find(EggItem).length).toBe(1);
    expect(renderedComponent.find(EggLink).length).toBe(1);
    expect(renderedComponent.find(EggLinkLabel).length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<EggItem />', () => {
  it('should render a Div', () => {
    const renderedComponent = shallow(<EggItem theme={theme} />);
    expect(renderedComponent.find('Div').length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<EggLink />', () => {
  it('should render a ClearLink', () => {
    const renderedComponent = shallow(<EggLink to="/#" theme={theme} />);
    expect(renderedComponent.find(ClearLink).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });
  it('should render its css w/ props fill', () => {
    const fill = '#000';
    const renderedComponent = shallow(
      <EggLink to="/#" theme={theme} fill={fill} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
  it('should render its css w/ props iconbg', () => {
    const iconbg = 'red';
    const renderedComponent = shallow(
      <EggLink to="/#" theme={theme} iconbg={iconbg} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
  it('should render its css w/o props border if not defined in theme', () => {
    const themeNoBorder = {
      radius: {
        buttons: '50px',
      },
      menu: {
        itemWidth: '32px',
        itemHeight: '32px',
      },
    };
    const renderedComponent = shallow(
      <EggLink
        to="/#"
        theme={themeNoBorder}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
