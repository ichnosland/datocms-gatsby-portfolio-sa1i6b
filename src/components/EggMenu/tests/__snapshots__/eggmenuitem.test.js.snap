// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EggItem /> should render a Div 1`] = `
.c0 {
  position: relative;
  height: 30px;
  width: 30px;
  border-radius: 100%;
  background-color: rgb(255,255,255);
  overflow: visible;
}

<Div
  className="c0"
  theme={
    Object {
      "menu": Object {
        "itemBorder": "2px solid green",
        "itemHeight": "32px",
        "itemWidth": "32px",
      },
      "radius": Object {
        "buttons": "50px",
      },
    }
  }
/>
`;

exports[`<EggLink /> should render a ClearLink 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  top: -1px;
  left: -2px;
  width: 32px;
  height: 32px;
  border: 2px solid green;
  background-color: rgb(255,255,255);
  outline: none;
  padding: 0;
  box-shadow: 0 1px 2px rgba(0,0,0,.3);
  -webkit-user-drag: none;
}

.c0 svg {
  max-width: 32px;
  max-height: 32px;
}

<ClearLink
  className="c0"
  theme={
    Object {
      "menu": Object {
        "itemBorder": "2px solid green",
        "itemHeight": "32px",
        "itemWidth": "32px",
      },
      "radius": Object {
        "buttons": "50px",
      },
    }
  }
  to="/#"
/>
`;

exports[`<EggLink /> should render its css w/ props fill 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  top: -1px;
  left: -2px;
  width: 32px;
  height: 32px;
  border: 2px solid green;
  background-color: rgb(255,255,255);
  outline: none;
  padding: 0;
  box-shadow: 0 1px 2px rgba(0,0,0,.3);
  -webkit-user-drag: none;
}

.c0 svg {
  max-width: 32px;
  max-height: 32px;
}

<ClearLink
  className="c0"
  fill="#000"
  theme={
    Object {
      "menu": Object {
        "itemBorder": "2px solid green",
        "itemHeight": "32px",
        "itemWidth": "32px",
      },
      "radius": Object {
        "buttons": "50px",
      },
    }
  }
  to="/#"
/>
`;

exports[`<EggLink /> should render its css w/ props iconbg 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  top: -1px;
  left: -2px;
  width: 32px;
  height: 32px;
  border: 2px solid green;
  background-color: red;
  outline: none;
  padding: 0;
  box-shadow: 0 1px 2px rgba(0,0,0,.3);
  -webkit-user-drag: none;
}

.c0 svg {
  max-width: 32px;
  max-height: 32px;
}

<ClearLink
  className="c0"
  iconbg="red"
  theme={
    Object {
      "menu": Object {
        "itemBorder": "2px solid green",
        "itemHeight": "32px",
        "itemWidth": "32px",
      },
      "radius": Object {
        "buttons": "50px",
      },
    }
  }
  to="/#"
/>
`;

exports[`<EggLink /> should render its css w/o props border if not defined in theme 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  top: -1px;
  left: -2px;
  width: 32px;
  height: 32px;
  border: none;
  background-color: rgb(255,255,255);
  outline: none;
  padding: 0;
  box-shadow: 0 1px 2px rgba(0,0,0,.3);
  -webkit-user-drag: none;
}

.c0 svg {
  max-width: 32px;
  max-height: 32px;
}

<ClearLink
  className="c0"
  theme={
    Object {
      "menu": Object {
        "itemHeight": "32px",
        "itemWidth": "32px",
      },
      "radius": Object {
        "buttons": "50px",
      },
    }
  }
  to="/#"
/>
`;

exports[`<EggMenuItem /> counts EggMenuItem children with empty url attribute 1`] = `
<EggMenuItem__EggItem>
  <EggMenuItem__EggLink
    onClick={[Function]}
    theme={
      Object {
        "brand": "#00BBEF",
        "menu": Object {
          "itemBorder": "none",
          "itemHeight": "32px",
          "itemWidth": "32px",
        },
        "radius": Object {
          "buttons": "6px",
        },
      }
    }
    to="#"
  >
    <Svg
      fill="#000"
      height="33px"
      path={
        <path
          d="M228 132.228c11.886-33.486 48.114-28 58.057 0 9.943 28-5.828 99.543-14.17 122.97-8.344 23.43-15.087 40-29.373 0-14.285-40-26.4-89.485-14.514-122.97zM256 320.8c16.343 0 29.486 13.143 29.486 29.486 0 16.342-13.143 29.485-29.486 29.485s-29.486-13.14-29.486-29.484c0-16.343 13.143-29.486 29.486-29.486zm.114 173.257c35.543 0 69.486-6.97 100.114-19.543.686-.343 1.37-.57 2.057-.914.114 0 .23 0 .23-.114 4.342-1.6 9.027-2.515 13.827-2.515 5.257 0 10.4 1.03 14.857 2.973L490.627 512l-27.2-108.8c0-6.514 1.83-12.686 4.8-17.943 1.03-1.6 1.943-3.2 3.086-4.57C497.144 342.17 512 296.342 512 247.314 512.228 110.63 397.6 0 256.114 0S0 110.63 0 246.973c0 136.57 114.628 247.086 256.114 247.086z"
          fillRule="evenodd"
        />
      }
      viewBox="0 0 512 512"
      width="33px"
    />
    <EggMenuItem__EggLinkLabel>
      Questo è un titolo
    </EggMenuItem__EggLinkLabel>
  </EggMenuItem__EggLink>
</EggMenuItem__EggItem>
`;

exports[`<EggMenuItem /> counts EggMenuItem children with url attribute 1`] = `
<EggMenuItem__EggItem>
  <EggMenuItem__EggLink
    onClick={[Function]}
    theme={
      Object {
        "brand": "#00BBEF",
        "menu": Object {
          "itemBorder": "none",
          "itemHeight": "32px",
          "itemWidth": "32px",
        },
        "radius": Object {
          "buttons": "6px",
        },
      }
    }
    to="/path"
  >
    <Svg
      fill="#000"
      height="33px"
      path={
        <path
          d="M228 132.228c11.886-33.486 48.114-28 58.057 0 9.943 28-5.828 99.543-14.17 122.97-8.344 23.43-15.087 40-29.373 0-14.285-40-26.4-89.485-14.514-122.97zM256 320.8c16.343 0 29.486 13.143 29.486 29.486 0 16.342-13.143 29.485-29.486 29.485s-29.486-13.14-29.486-29.484c0-16.343 13.143-29.486 29.486-29.486zm.114 173.257c35.543 0 69.486-6.97 100.114-19.543.686-.343 1.37-.57 2.057-.914.114 0 .23 0 .23-.114 4.342-1.6 9.027-2.515 13.827-2.515 5.257 0 10.4 1.03 14.857 2.973L490.627 512l-27.2-108.8c0-6.514 1.83-12.686 4.8-17.943 1.03-1.6 1.943-3.2 3.086-4.57C497.144 342.17 512 296.342 512 247.314 512.228 110.63 397.6 0 256.114 0S0 110.63 0 246.973c0 136.57 114.628 247.086 256.114 247.086z"
          fillRule="evenodd"
        />
      }
      viewBox="0 0 512 512"
      width="33px"
    />
    <EggMenuItem__EggLinkLabel>
      Questo è un titolo
    </EggMenuItem__EggLinkLabel>
  </EggMenuItem__EggLink>
</EggMenuItem__EggItem>
`;

exports[`<EggMenuItem /> counts EggMenuItem with one child and no url attribute 1`] = `<EggMenuItem__EggItem />`;
